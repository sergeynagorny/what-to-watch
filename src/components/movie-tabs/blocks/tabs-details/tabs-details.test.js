import React from "react";
import renderer from "react-test-renderer";
import TabsDetails from "./tabs-details.jsx";


const mock = {
  film: {
    director: [`Wes Andreson`],
    starring: [`Bill Murray`, `Edward Norton`, `Jude Law`, `Willem Dafoe`, `Saoirse Ronan`, `Tony Revoloru`, `Tilda Swinton`, `Tom Wilkinson`],
    genre: [`Comedy`],
    released: new Date(`2014-2-6`),
    runtime: 112,
  },
};

it(`Should TabsDetails render correctly`, () => {
  const {film, reviews} = mock;

  const tree = renderer.create(
      <TabsDetails
        film={film}
        reviews={reviews}
      />
  ).toJSON();

  expect(tree).toMatchSnapshot();
});
