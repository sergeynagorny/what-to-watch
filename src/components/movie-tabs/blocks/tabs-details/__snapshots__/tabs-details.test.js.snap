// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should TabsDetails render correctly 1`] = `
<div
  className="movie-card__text movie-card__row"
>
  <div
    className="movie-card__text-col"
  >
    <p
      className="movie-card__details-item"
    >
      <strong
        className="movie-card__details-name"
      >
        Director
      </strong>
      <span
        className="movie-card__details-value"
        style={
          Object {
            "whiteSpace": "pre-line",
          }
        }
      >
        Wes Andreson
      </span>
    </p>
    <p
      className="movie-card__details-item"
    >
      <strong
        className="movie-card__details-name"
      >
        Starring
      </strong>
      <span
        className="movie-card__details-value"
        style={
          Object {
            "whiteSpace": "pre-line",
          }
        }
      >
        Bill Murray, 
Edward Norton, 
Jude Law, 
Willem Dafoe, 
Saoirse Ronan, 
Tony Revoloru, 
Tilda Swinton, 
Tom Wilkinson
      </span>
    </p>
  </div>
  <div
    className="movie-card__text-col"
  >
    <p
      className="movie-card__details-item"
    >
      <strong
        className="movie-card__details-name"
      >
        Run Time
      </strong>
      <span
        className="movie-card__details-value"
        style={
          Object {
            "whiteSpace": "pre-line",
          }
        }
      >
        01h 52m 
      </span>
    </p>
    <p
      className="movie-card__details-item"
    >
      <strong
        className="movie-card__details-name"
      >
        Genre
      </strong>
      <span
        className="movie-card__details-value"
        style={
          Object {
            "whiteSpace": "pre-line",
          }
        }
      >
        Comedy
      </span>
    </p>
    <p
      className="movie-card__details-item"
    >
      <strong
        className="movie-card__details-name"
      >
        Released
      </strong>
      <span
        className="movie-card__details-value"
        style={
          Object {
            "whiteSpace": "pre-line",
          }
        }
      >
        2014
      </span>
    </p>
  </div>
</div>
`;
